docs/web/concepts-layout.mdx

```diff
@@ -0,0 +1,303 @@
+When a feature in the results list is opened, the `<feature-details>` component will be activated. At this point, the panel takes over the activation of the component. It will display the new component overtop of the existing visible component, and provide a back button to navigate to the previous component. The activation and deactivation algorithms for a panel rely on the concept of a stacking order of components laid on top of eachother, with the first component being the bottomost, and the last being the topmost. In the example above, the `<stack>` containing `<search>` is the lowest, while the `<feature-details>` is the topmost.
+
+### Activation
+
+-   If the target component exists in the panel, then deactivate components above it until the target component is active
+    -   Else, the target component must be deactivated, so activate it and place it above the existing component.
+
+### Deactivation
+
+-   If the component is the top most element (it is active), then deactivate it.
+    -   Else, the component is already inactive
+
+## Models
+
+Components often observe and modify the state of other components.
+Consider the following example:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<layout xmlns="https://geocortex.com/layout/v1">
+    <map id="mapA"/>
+    <map id="mapB"/>
+    <panel>
+        <stack>
+            <scalebar id="A" />
+            <scalebar id="B" />
+        </stack>
+    </panel>
+</layout>
+```
+
+Our scale bar component displays information about a particular map. In our example, scale bars A and B are expected to show the correct scale based on their context.
+
+Layouts achieve these simply by expressing data dependencies in the form of models that are marked as "imported" or "exported". If we know that the scale bar "imports" a particular type of model, and that the map "exports" the same type of model, we can resolve those dependencies simply by plugging exported models into the components that import them.
+
+```ts
+@exportModel
+export class MapComponent {
+    // ... map stuff ...
+}
+
+export class ScaleBarComponent {
+    @importModel("map-extension")
+    map: MapExtension;
+
+    // ... scale bar stuff ...
+}
+```
+
+When the system slots each scale bar into the panel, it will attempt to satisfy any model imports declared by the component being slotted.
+
+By default, this happens by walking up the tree from the location of the component being slotted and simply finding the first ancestor that exports a model with the matching type, like in this basic example;
+
+```xml
+<map>
+    <scalebar/>
+</map>
+```
+
+However, in the previous example with two scale bars, the scale bars will walk up to the root of the layout and not locate any maps. To solve this, we can bind each scale bar to the appropiate map with the `models` attribute.
```

Comment:
The above statement in the docs still is true, in that they will not locate any maps walking up the root. I just don't mention the fallback behavior of a breadth first search.

I felt that behavior was advanced, so I included under the advanced layout topic. Do you think it's worth  to that here with a quick mention or :::note?

https://deploy-preview-16--geocortex-dev-center.netlify.app/docs/web/concepts-advanced-layout#nested-model-imports

Reply:
Yeah, I saw that since writing this comment. I agree that this isn't the place to be super precise about the exact algorithm that's used. But I find "not locate any maps" very misleading. The problem to be solved in this example is how to match scale bar A with map A, and scale bar B with map B, which is solved by adding explicit `models=` attributes. Something along those lines would be more accurate without going into details about tree walking and breadth-first searches.

Reply:
Good points. Changed the wording to be more explicit